Source/SpectrumAnalysis/KTSequentialTrackFinder.cc

```diff
@@ -0,0 +1,502 @@
+            fDoneSlot("done", this, &KTSequentialTrackFinder::AcquisitionIsOver, &fClusterDoneSignal)
+    {
+    }
+
+    KTSequentialTrackFinder::~KTSequentialTrackFinder()
+    {
+    }
+
+    bool KTSequentialTrackFinder::Configure(const scarab::param_node* node)
+    {
+        if (node == NULL) return false;
+
+        SetSNRPowerThreshold(node->get_value("snr-threshold-power", GetSNRPowerThreshold()));
+        SetTrimmingFactor(node->get_value("trimming-factor", GetTrimmingFactor()));
+        SetLinePowerWidth(node->get_value("half-line-width", GetLinePowerWidth()));
+        SetTimeGapTolerance(node->get_value("time-gap-tolerance", GetTimeGapTolerance()));
+        SetFrequencyAcceptance(node->get_value("frequency-acceptance", GetFrequencyAcceptance()));
+        SetInitialSlope(node->get_value("initial-slope", GetInitialSlope()));
+        SetMinPoints(node->get_value("min-points", GetMinPoints()));
+        SetMinSlope(node->get_value("min-slope", GetMinSlope()));
+
+        SetMinFrequency(node->get_value("min-frequency", GetMinFrequency()));
+        SetMaxFrequency(node->get_value("max-frequency", GetMaxFrequency()));
+
+
+        if (node->has("apply-power-cut"))
+        {
+            SetApplyPowerCut(node->get_value("apply-power-cut", GetApplyPowerCut()));
+            SetPowerThreshold(node->get_value("power-threshold", GetPowerThreshold()));
+        }
+        if (node->has("apply-power-density-cut"))
+        {
+            SetApplyDensityCut(node->get_value("apply-power-density-cut", GetApplyDensityCut()));
+            SetDensityThreshold(node->get_value("power-density-threshold", GetDensityThreshold()));
+        }
+        if (node->has("min-bin"))
+        {
+            SetMinBin(node->get_value< unsigned >("min-bin"));
+            SetCalculateMinBin(false);
+        }
+        if (node->has("max-bin"))
+        {
+            SetMaxBin(node->get_value< unsigned >("max-bin"));
+            SetCalculateMaxBin(false);
+        }
+        if (node->has("minimum-line-bin-distance"))
+		{
+        	SetMinFreqBinDistance(node->get_value<double>("minimum-line-bin-distance"));
+		}
+        if (node->has("search-radius"))
+        {
+        	SetSearchRadius(node->get_value<int>("search-radius"));
+        }
+        if (node->has("converge-delta"))
+        {
+            SetConvergeDelta(node->get_value<int>("converge_delta"));
+        }
+
+
+
+        return true;
+    }
+
+    bool KTSequentialTrackFinder::SetPreCalcGainVar(KTGainVariationData& gvData)
+    {
+        fGVData = gvData;
+        return true;
+    }
+
+    bool KTSequentialTrackFinder::RunSequentialTrackFinding(KTSliceHeader& slHeader, KTPowerSpectrumData& spectrum)
+    {
+        this->PointLineAssignment(slHeader, spectrum, fGVData);
+        return true;
+    }
+
+    bool KTSequentialTrackFinder::PointLineAssignment(KTSliceHeader& slHeader, KTPowerSpectrumData& spectrum, KTGainVariationData& gvData)
```

Comment:
I added a three-data-type slot class in Nymph called KTSlotDataThreeTypes, which you should be able to use here.  This way you won't need to do thresholding here, and you can instead use already-discriminated points.  It's currently in the branch feature/threetypes.  Let me know if it works, and I can make a Nymph release.